src/main/java/com/gooddata/executeafm/afm/PopMeasureDefinition.java

```diff
@@ -48,11 +58,20 @@ public PopMeasureDefinition(@JsonProperty("measureIdentifier") final String meas
     }
 
     @Override
-    public MeasureDefinition withObjUriQualifier(final UriObjQualifier qualifier) {
-        return new PopMeasureDefinition(measureIdentifier, qualifier);
+    public MeasureDefinition withObjUriQualifiers(final ObjQualifierConverter objQualifierConverter) {
+        if (popAttribute instanceof IdentifierObjQualifier) {
+            final IdentifierObjQualifier identifierPopAttribute = (IdentifierObjQualifier) popAttribute;
+            Validate.notNull(objQualifierConverter, "'objQualifierConverter' must not be null!");
+            return objQualifierConverter.convertToUriQualifier(identifierPopAttribute)
+                    .map(uriObjQualifier -> new PopMeasureDefinition(measureIdentifier, uriObjQualifier))
+                    .orElseThrow(() -> new IllegalArgumentException(format("The converter does not know how to convert identifier '%s'!", popAttribute)));
+        }
+        return this;
```

Comment:
the method contract is stated to return a copy, to be honest I am not sure why, may be the contract can be changed then if the use case doesn't require the copy always.

Reply:
I updated the javadoc. The `@return` tag described the correct behavior but the main description was misleading.